equinity-historical-data/src/test/java/uk/gov/justice/laa/crime/equinity/historicaldata/controller/report/Crm5UpperLimitReportControllerTest.java

```diff
@@ -0,0 +1,176 @@
+package uk.gov.justice.laa.crime.equinity.historicaldata.controller.report;
+
+import jakarta.validation.ConstraintViolationException;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.InvalidDataAccessResourceUsageException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.justice.laa.crime.equinity.historicaldata.exception.DateRangeConstraintViolationException;
+import uk.gov.justice.laa.crime.equinity.historicaldata.exception.ResourceNotFoundException;
+import uk.gov.justice.laa.crime.equinity.historicaldata.service.report.Crm5UpperLimitReportService;
+
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+import static uk.gov.justice.laa.crime.equinity.historicaldata.service.CrmFileService.CRM_TYPE_5;
+
+@SpringBootTest
+@ExtendWith(SoftAssertionsExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ActiveProfiles("local")
+class Crm5UpperLimitReportControllerTest {
+    private List<String> testInvalidFormatDates;
```

Comment:
I know its a test class, but nice to have one, Just a suggestion
move `private List<String> testInvalidFormatDates;` after `private static final` (after line 35)

Reply:
sorted